pkg/run/run.go

```diff
@@ -6,44 +6,51 @@ import (
 	"github.com/pkg/errors"
 	"golang.stackrox.io/kube-linter/internal/version"
 	"golang.stackrox.io/kube-linter/pkg/checkregistry"
+	"golang.stackrox.io/kube-linter/pkg/config"
 	"golang.stackrox.io/kube-linter/pkg/diagnostic"
 	"golang.stackrox.io/kube-linter/pkg/ignore"
 	"golang.stackrox.io/kube-linter/pkg/instantiatedcheck"
 	"golang.stackrox.io/kube-linter/pkg/lintcontext"
 )
 
-type checkStatus string
+// CheckStatus is enum type.
+type CheckStatus string
 
 const (
-	checksPassed checkStatus = "Passed"
-	checksFailed checkStatus = "Failed"
+	// ChecksPassed means no lint errors found.
+	ChecksPassed CheckStatus = "Passed"
+	// ChecksFailed means lint errors were found.
+	ChecksFailed CheckStatus = "Failed"
 )
 
 // Result represents the result from a run of the linter.
 type Result struct {
+	Checks  []config.Check
 	Reports []diagnostic.WithContext
 	Summary Summary
 }
 
 // Summary holds information about the linter run overall.
 type Summary struct {
-	ChecksStatus      checkStatus
-	CheckTime         time.Time
+	ChecksStatus      CheckStatus
+	CheckEndTime      time.Time
```

Comment:
Hmm this is a backwards-incompatible change, both for library users and JSON output users. I would suggest to just keep it as `CheckTime`.

Reply:
Yes, it is breaking change but JSON output was merged only 9 days ago https://github.com/stackrox/kube-linter/commit/f7e6f6e7941eca677299b3007626664272ffc9f0#diff-bb95634405f8d047841128dbd79e9058b927a35a23011aa7096d899c21764896R31
I hope there are not many users of `CheckTime` specifically.

Reply:
Hmm, I doubt there are too, but in my experience, you can sometimes be surprised by how users start adopt specific things. In any case, I'm okay with the risk here.